lib/kafka/connection.rb

```diff
@@ -156,8 +227,8 @@ def write_request(request, notification)
     #   a given Decoder.
     #
     # @return [nil]
-    def read_response(response_class, notification)
-      @logger.debug "Waiting for response #{@correlation_id} from #{to_s}"
+    def read_response(expected_correlation_id, notification)
```

Comment:
Yup, at least that's the design. I ran this in production for a good hour and didn't see any problems, but I'm going to be doing at least one beta release before a final.